src/app/screens/Accounts/ImportSecretKey/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { CopyIcon } from "@bitcoin-design/bitcoin-icons-react/filled";
+import Container from "@components/Container";
+import Loading from "@components/Loading";
+import * as bip39 from "@scure/bip39";
+import { wordlist } from "@scure/bip39/wordlists/english";
+import { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useParams } from "react-router-dom";
+import { toast } from "react-toastify";
+import Alert from "~/app/components/Alert";
+import Button from "~/app/components/Button";
+import MnemonicInputs from "~/app/components/MnemonicInputs";
+import { useAccount } from "~/app/context/AccountContext";
+import { saveMnemonic } from "~/app/utils/saveMnemonic";
+import msg from "~/common/lib/msg";
+
+function ImportSecretKey() {
+  const [mnemonic, setMnemonic] = useState<string>("");
+  const account = useAccount();
+  const { t: tCommon } = useTranslation("common");
+  const [importPasteLabel, setImportPasteLabel] = useState(
+    tCommon("actions.paste_clipboard") as string
+  );
+  const { t } = useTranslation("translation", {
+    keyPrefix: "accounts.account_view.mnemonic",
+  });
+
+  // TODO: useMnemonic hook
+  const [hasFetchedData, setHasFetchedData] = useState(false);
+  const [hasMnemonic, setHasMnemonic] = useState(false);
+  // TODO: useNostrPrivateKey hook
+  const [currentPrivateKey, setCurrentPrivateKey] = useState("");
+  const { id } = useParams();
+
+  const fetchData = useCallback(async () => {
+    try {
+      if (id) {
+        const priv = (await msg.request("nostr/getPrivateKey", {
+          id,
+        })) as string;
+        if (priv) {
+          setCurrentPrivateKey(priv);
+        }
+
+        const accountMnemonic = (await msg.request("getMnemonic", {
+          id,
+        })) as string;
+        if (accountMnemonic) {
+          setHasMnemonic(true);
+        }
```

Comment:
can these calls somehow be combined? to get one state from the background?

Reply:
I am not sure. The nostr key is not necessarily related to the mnemonic if it is imported. Would we add a new method like getKeys() or something which returns all the different private and public keys as well as the mnemonic?

Reply:
both data are now available from `await api.getAccount()` (`nostrEnabled`, `hasMnemonic`)

"getMnemonic" is only requested in one place now - the backup screen 🎉